atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java

```diff
@@ -378,6 +379,31 @@ public InetSocketAddress apply(String endpoint) {
             return client.describe_ring(config.keyspace());
         }};
 
+    // todo add interior retry? currently can't do this with good code re-use because runWithRetryOnHost is too smart and can shift hosts, which we don't want here
+    public <V, K extends Exception> V runOnAllNodesAndEnsureReturnValuesConsistent(FunctionCheckedException<Cassandra.Client, V, K> f) throws IllegalStateException {
+            Map<InetSocketAddress, Exception> exceptionsByHost = Maps.newHashMap();
+            Map<InetSocketAddress, V> returnValuesByHost = Maps.newHashMap();
+
+            for (Entry<InetSocketAddress, CassandraClientPoolingContainer> poolForHost : currentPools.entrySet()) {
+                try {
+                    returnValuesByHost.put(poolForHost.getKey(), poolForHost.getValue().runWithPooledResource(f));
+                } catch (Exception e) {
+                    exceptionsByHost.put(poolForHost.getKey(), e);
+                }
+            }
+
+        ImmutableSet<V> returnValueSet = ImmutableSet.copyOf(returnValuesByHost.values());
```

Comment:
this assumes that the values implement hashCode/equals properly, which is fine for case of CF ID `Integer`, but probably needs to be a warning on the method javadoc